lib/Dialect/TTNN/Utils/TransformUtils.cpp

```diff
@@ -31,4 +32,60 @@ Value getOrInsertDevice(PatternRewriter &rewriter, Operation *op) {
   rewriter.restoreInsertionPoint(currentInsertionPoint);
   return deviceOp.getResult();
 }
+
+// Helper method to insert a ToLayoutOp to convert the input operand to the
+// desired tensor layout, buffer type and memory layout.
+mlir::Value
+createToLayoutOp(Operation *op, Value inputValue, PatternRewriter &rewriter,
+                 Layout targetTensorLayout, BufferType targetTensorBufferType,
+                 std::optional<TensorMemoryLayout> targetTensorMemoryLayout) {
+  TTNNLayoutAttr inputLayoutAttr = getLayoutAttrFromValue(inputValue);
+
+  // Create element type based on tensor layout.
+  Type elementType = getElementType(rewriter.getContext(), targetTensorLayout,
+                                    inputLayoutAttr.getDataType());
+
+  // Create tensor memory layout attribute.
+  ttnn::TensorMemoryLayoutAttr outputMemLayoutAttr =
+      targetTensorMemoryLayout.has_value()
+          ? ttnn::TensorMemoryLayoutAttr::get(rewriter.getContext(),
+                                              targetTensorMemoryLayout.value())
+          : nullptr;
+
+  // Create the output memory config attribute.
+  ttnn::MemoryConfigAttr outputMemConfigAttr = ttnn::MemoryConfigAttr::get(
+      rewriter.getContext(),
+      ttnn::BufferTypeAttr::get(rewriter.getContext(), targetTensorBufferType),
+      ttnn::ShardSpecAttr::get(
+          op->getContext(),
+          ttnn::ShapeAttr::get(rewriter.getContext(),
+                               inputLayoutAttr.getMemref().getShape())),
+      outputMemLayoutAttr);
+
+  // Get the input operand type.
+  RankedTensorType inputOperandType =
+      mlir::cast<RankedTensorType>(inputValue.getType());
+
+  // Create a ToLayoutOp to convert the input operand to the desired
+  // tensor layout, buffer type and memory layout.
+  return rewriter
+      .create<ttnn::ToLayoutOp>(
+          op->getLoc(),
+          ttnn::utils::createRankedTensorTypeWithEncoding(
+              inputOperandType,
+              inputLayoutAttr
+                  .withElementType(rewriter.getContext(), elementType)
+                  .withBufferType(rewriter.getContext(), targetTensorBufferType)
+                  .withMemoryLayout(rewriter.getContext(),
+                                    outputMemLayoutAttr)),
```

Comment:
Can you move this out of this function call for readability.

Reply:
Same for this

(targetTensorBufferType == ttnn::BufferType::SystemMemory)
              ? nullptr
              : utils::getOrInsertDevice(rewriter, op)